.github/copilot-instructions.md

```diff
@@ -0,0 +1,334 @@
+# nvm Copilot Instructions
+
+This document provides guidance for GitHub Copilot when working with the Node Version Manager (nvm) codebase.
+
+## Overview
+
+nvm is a version manager for Node.js, implemented as a POSIX-compliant function that works across multiple shells (sh, dash, bash, ksh, zsh). The codebase is primarily written in shell script and emphasizes portability and compatibility.
+
+### Core Architecture
+
+- **Main script**: `nvm.sh` - Contains all core functionality and the main `nvm()` function
+- **Installation script**: `install.sh` - Handles downloading and installing nvm itself
+- **Execution wrapper**: `nvm-exec` - Allows running commands with specific Node.js versions
+- **Bash completion**: `bash_completion` - Provides tab completion for bash users
+- **Tests**: Comprehensive test suite in `test/` directory using the urchin test framework
+
+## Key Files and Their Purposes
+
+### `nvm.sh`
+The core functionality file containing:
+- Main `nvm()` function (starts around line 3000)
+- All internal helper functions (prefixed with `nvm_`)
+- Command implementations for install, use, ls, etc.
+- Shell compatibility logic
+- POSIX compliance utilities
+
+### `install.sh`
+Handles nvm installation via curl/wget:
+- Downloads nvm from GitHub
+- Sets up directory structure
+- Configures shell integration
+- Supports both git clone and script download methods
+
+### `nvm-exec`
+Simple wrapper script that:
+- Sources nvm.sh with `--no-use` flag
+- Switches to specified Node version via `NODE_VERSION` env var or `.nvmrc`
+- Executes the provided command with that Node version
+
+## Top-Level nvm Commands and Internal Functions
+
+### Core Commands
+
+#### `nvm install [version]`
+- **Internal functions**: `nvm_install_binary()`, `nvm_install_source()`, `nvm_download_artifact()`
+- Downloads and installs specified Node.js version
+- Supports LTS versions, version ranges, and aliases
+- Can install from binary or compile from source
+
+#### `nvm use [version]`
+- **Internal functions**: `nvm_resolve_alias()`, `nvm_version_path()`, `nvm_change_path()`
+- Switches current shell to use specified Node.js version
+- Updates PATH environment variable
+- Supports `.nvmrc` file integration
+
+#### `nvm ls [pattern]`
+- **Internal functions**: `nvm_ls()`, `nvm_tree_contains_path()`
+- Lists installed Node.js versions
+- Supports pattern matching and filtering
+- Shows current version and aliases
+
+#### `nvm ls-remote [pattern]`
+- **Internal functions**: `nvm_ls_remote()`, `nvm_download()`, `nvm_ls_remote_index_tab()`
+- Lists available Node.js versions from nodejs.org
+- Supports LTS filtering and pattern matching
+- Downloads version index on-demand
+
+#### `nvm alias [name] [version]`
+- **Internal functions**: `nvm_alias()`, `nvm_alias_path()`
+- Creates symbolic links for version aliases
+- Special aliases: `default`, `node`, `stable`, `unstable`
```

Comment:
also `iojs`. also note that `stable` and `unstable` are deprecated, since they're from node's pre-v1 release plan